tasks/internal/update-fbc-catalog-task/update-fbc-catalog-task.yaml

```diff
@@ -126,6 +126,14 @@ spec:
             jq --arg fbc_fragment "${fbc_fragment}" \
               '[.items[] |select(.fbc_fragment==$fbc_fragment and .state!="failed")][0] // empty')
 
+          # if a previous build is found, but from_index and index_image are different, the result of the next
+          # expression should be different than 1, meaning that the previous build was a staging index build,
+          # then we should return here, causing the task to call IIB again, but with overwrite_from_index set
+          # to true.
+          if [ "$(jq '[ .from_index, .index_image ] | unique | length' <<< "${build}")" != 1 ]; then
```

Comment:
Why not simply check if they are equal?
```suggestion
          if [ "$(jq '.from_index == .index_image' <<< "${build}")" != true ]; then
```

Reply:
Yes. That makes more sense. I just adjusted as well the comparison to check if `.from_index != .index_image`  equals `true`  facilitate the understand following the logic flow (at least for me :D).